wa/workloads/speedometer/__init__.py

```diff
@@ -12,57 +12,295 @@
+        ) as read_fh:
+            lines = read_fh.readlines()
+
+            # Add additional elements for the preferences XML to the
+            # _second-last_ line
+            for line in [
+                '<boolean name="first_run_flow" value="true" />\n',
+                '<boolean name="first_run_tos_accepted" value="true" />\n',
+                '<boolean name="first_run_signin_complete" value="true" />\n',
+            ]:
+                lines.insert(len(lines) - 1, line)
+
+            with open(
+                os.path.join(context.output_directory, self.preferences_xml + ".new"),
+                "w",
+            ) as write_fh:
+                for line in lines:
+                    write_fh.write(line)
+
+        # Make sure ownership of the original file is preserved.
+        user_owner, group_owner = self.target.execute(
+            "ls -l {}".format(file_to_modify),
+            as_root=True,
+        ).split()[2:4]
+
+        self.target.push(
+            os.path.join(context.output_directory, self.preferences_xml + ".new"),
+            file_to_modify,
+            as_root=True,
+        )
+
+        self.target.execute(
+            "chown {}.{} {}".format(user_owner, group_owner, file_to_modify),
+            as_root=True,
+        )
+
+    def run(self, context):
+        super(Speedometer, self).run(context)
+
+        browser_launch_cmd = "am start -a android.intent.action.VIEW -d {} {}".format(
+            self.speedometer_url, self.chrome_package
+        )
+        self.target.execute(browser_launch_cmd)
+
+        # Wait 60 seconds before we start polling the system to check which
+        # processes are busy.
+        time.sleep(60)
+
+        # We can only consider the benchmark complete and proceed if we see 5
+        # times _in a row_ that a process containing the string
+        # 'sandboxed_process' is not the hottest process in the system. We will
+        # sleep for 2 seconds between every check.
+        top_cmd = "{} top -b -n1".format(self.target.busybox)
+
+        def _top_output_for_busiest_process():
+            # busybox's top always outputs the Mem/CPU/Load/Header lines first,
+            # so line 5 is the hottest process.
+            return self.target.execute(top_cmd).split("\n")[4]
+
+        def _get_all_processes():
+            return self.target.execute(top_cmd).split("\n")[4:]
+
+        search_term = self.chrome_package + ":sandbox"
+        if self.render_process_search_term is not None:
+            search_term = self.render_process_search_term
+        render_processes = [x for x in _get_all_processes() if search_term in x]
+        if len(render_processes) == 0:
+            raise WorkloadError(
+                "Unable to find any suitable render process. Try setting "
+                "'render_process_search_term'."
+            )
+
+        countdown = 5
+        while countdown > 0:
+            while "sandboxed_process" in _top_output_for_busiest_process():
```

Comment:
This doesn't appear to use the `search_term` you've checked for?

Reply:
Good catch! :) fixed.